xarray/conventions.py

```diff
@@ -502,6 +507,20 @@ def stackable(dim):
                     new_vars[k].encoding["coordinates"] = coord_str
                     del var_attrs["coordinates"]
                     coord_names.update(var_coord_names)
+            if "bounds" in var_attrs:
```

Comment:
This new behaviour is controlled by `decode_coords` which is True by default. To deal with backward compatibility, should we add `decode_aux_vars` that controls whether these auxiliary variables are set as coords?

Reply:
I feel like that question is above my pay grade.

Would that also cover `ancillary_variables` and `cell_measures` if someone decides they want those to be coords instead of data vars?

Reply:
[There's also a `decode_cf` argument to `open_dataset`](https://xarray.pydata.org/en/stable/generated/xarray.open_dataset.html#xarray.open_dataset), which is another possibility for the argument controlling the behavior.

I have the loop over attributes I think are lists of variables, including `ancillary_variables` and `cell_measures`.

Reply:
I agree that `decode_cf` should control this since we are decoding CF attributes.

Reply:
On further inspection, [`decode_cf` appears to be a quick way to control `mask_and_scale`, `decode_coords`, `decode_times`, `decode_timedelta`, and `concat_characters`](https://github.com/DWesl/xarray/blob/ca0f80523ef84f9159c0eebf2580354dd7459d75/xarray/backends/api.py#L437), so, in a sense, the new behavior can already be turned off by `decode_cf`.

Whether the new behavior should remain under the control of the existing `decode_coords` or a new `decode_aux_vars`.
This is primarily sticking things in `ds.coords` if the file says they should be, the difference is whether to look only at the `coordinates` attribute or at more attributes from the CF standard.